Makefile

```diff
@@ -53,7 +53,7 @@ GO_LINKER_FLAGS ?= -ldflags \
 				    -X 'github.com/mattermost/mattermost-server/model.BuildDate=$(BUILD_DATE)'\
 				    -X github.com/mattermost/mattermost-server/model.BuildHash=$(BUILD_HASH)\
 				    -X github.com/mattermost/mattermost-server/model.BuildHashEnterprise=$(BUILD_HASH_ENTERPRISE)\
-				    -X github.com/mattermost/mattermost-server/model.BuildEnterpriseReady=$(BUILD_ENTERPRISE_READY)"
+				    -X github.com/mattermost/mattermost-server/model.BuildEnterpriseReady=$(BUILD_ENTERPRISE_READY) $(LDFLAGS)"
```

Comment:
If we keep as-is, let's pull this down onto it's own line, but I'm wondering if it's weird to support both `$(LDFLAGS)` /and/ continue to allow `GO_LINKER_FLAGS` to be overridden -- would the existing ability to override suit your use case?

Separately -- and I don't have an answer to this -- is there any reason someone might have exported `$LDFLAGS` into their environment such that this change would cause unexpected build changes? (I'm not sure if exporting `$LDFLAGS` is a "thing", per se.)

Reply:
Ideally I would prefer to do like TiDB's [Makefile](https://github.com/pingcap/tidb/blob/master/Makefile): `LDFLAGS += -X ...`.
But this means having to preprend `-ldflags` everywhere, thus having `-ldflags $(LDFLAGS)`
If `GO_LINKER_FLAGS` is only set here, it could be replaced by `LDFLAGS` (or vice-versa)

Reply:
For exporting `LDFLAGS`, it depends on the internal tooling. If there are shell scripts using `$LDFLAGS`, like with `autoconf`, yes it can modify the build. That's a way to change the compiler/linker behavior. Here, I haven't seen this variable used.

I prefer `LDFLAGS` over `GO_LINKER_FLAGS` (some also name it `CGO_LDFLAGS`), because it's a _standard_ naming used by linkers, and the arguments it takes are quite standard too.
More ever, the argument is `-ldflags`.

Reply:
You make a good case to adopt a standard naming: I'd support renaming `GO_LINKER_FLAGS` to `LDFLAGS`, and optionally redefining it with the `-ldflags` prefix to simplify usage where it's needed.